### PR TITLE
Add error handling code for some external APIs

### PR DESCRIPTION
**Pull request by ZhouyangJia**
Hi,

I analyzed the source code and find some APIs missing error handling code, which may cause crashes. This change is aimed to fix the potential bugs.

Best,
Zhouyang
## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Your changes looks very good and make really sence. Thanks a lot!

Only try to change directly to new C++11 style for the `NULL` with `nullptr` and for `bind(testSocket, (struct sockaddr*)&m_addr.saddr, m_addr.size)` with `bind(testSocket, static_cast<struct sockaddr*>(&m_addr.saddr), m_addr.size)`.

Question for the others in team, makes it sence to add a `CLog::Log(...)` message there, to see the source of fault?

File: xbmc/network/Socket.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
           for (m_iPort = port; m_iPort <= port + range; ++m_iPort)
           {
             m_addr.saddr.saddr6.sin6_port = htons(m_iPort);
-            if (bind(testSocket, (struct sockaddr*)&m_addr.saddr, m_addr.size) >= 0)
+            if (bind(testSocket, static_cast<struct sockaddr*>(&m_addr.saddr), m_addr.size) >= 0)
```
Comment (AlwinEsch) [inline xbmc/network/Socket.cpp]: Sorry, has testet for my comment now. Should be `reinterpret_cast` otherwise it brings compile fault.

**Comment by Razzeee** [comment]
@AlwinEsch yes, I would encourage adding some logging

**Comment by AlwinEsch** [comment]
From my side looks OK to merge and become in (only the NULL still there ;-) )

**Comment by phunkyfish** [comment]
@ZhouyangJia do you want to fix the final comment from @AlwinEsch so this can be merged?

**Comment by AlwinEsch** [comment]
Yes, the NULL and also a bit logging. To have this fix request whould be good.

**Comment by ZhouyangJia** [comment]
Hi, I fixed these issues in #17458 

Sorry for the late reply.

**Pull request closed without merging**